cpp/ql/src/semmle/code/cpp/Type.qll

```diff
@@ -301,7 +310,14 @@ class BuiltInType extends Type, @builtintype {
 }
 
 /**
- * An erroneous type.
+ * An erroneous type.  This type has no corresponding C/C++ syntax.
+ *
+ * `ErroneousType` is the type of `ErrorExpr`, which in turn refers to an illegal
+ * language construct.  In the example below, a temporary (`0`) cannot be bound
+ * to an lvalue reference (`int &`):
+ * ```
+ * int &intref = 0;
```

Comment:
Have we actually checked that something of `ErroneousType` appears in this example?

Reply:
Added a test: https://github.com/Semmle/ql/pull/2315

Reply:
(yes it does)